Meadowlark-js/backends/meadowlark-aws-dynamodb-opensearch/src/BaseDynamoRepository.ts

```diff
@@ -63,74 +81,66 @@ export function assignableSortKeyFromId(id: string): string {
   return `ASSIGN#ID#${id}`;
 }
 
-export function sortKeyFromEntityIdentity(entityInfo: EntityInfo): string {
-  return sortKeyFromId(documentIdForEntityInfo(entityInfo));
+export function sortKeyFromEntityIdentity(documentInfo: DocumentInfo): string {
+  return sortKeyFromId(documentIdForEntityInfo(documentInfo));
 }
 
-export function conditionCheckFrom(entityIdentifyingInfo: EntityIdentifyingInfo) {
-  invariant(
-    entityIdentifyingInfo.naturalKey.startsWith('NK#'),
-    `entityIdentifyingInfo.naturalKey "${entityIdentifyingInfo.naturalKey}" did not start with "NK#"`,
-  );
+export function conditionCheckFrom(entityIdentifyingInfo: DocumentIdentifyingInfo) {
   return {
     ConditionCheck: {
       TableName: tableOpts.tableName,
       Key: {
         pk: entityTypeStringFrom(entityIdentifyingInfo),
-        sk: sortKeyFromId(documentIdFromNaturalKeyString(entityIdentifyingInfo.naturalKey)),
+        sk: sortKeyFromId(idFromDocumentElements(entityIdentifyingInfo.documentIdentity)),
```

Comment:
renaming "natural key" (a relational db concept) to "identity" in our Meadowlark ubiquitous language

Reply:
Good catch, the name should be `idFromDocumentIdentity`. 😂